jetty-annotations/src/main/java/org/eclipse/jetty/annotations/WebFilterAnnotation.java

```diff
@@ -108,7 +108,7 @@ public void apply()
 
             FilterMapping mapping = new FilterMapping();
             mapping.setFilterName(holder.getName());
-
+            metaData.setOrigin(name + ".filter.mapping." + mapping.hashCode(), filterAnnotation, clazz);
```

Comment:
It is good to add the hashCode, but can we do it with with @hex rather than .long
```suggestion
            metaData.setOrigin(String.format("%s.filter.mapping@%x", name, mapping.hashCode()), filterAnnotation, clazz);
```

Reply:
hmmm can @ be put into an XML comment???

Reply:
The origin is used programmatically to determine whether or not we're allowed to add more mappings, change mappings etc. It actually doesn't have anything to do with the output of the "origin" attribute for quickstart, which is simply the url of the descriptor that declared the mapping, or the classname of the class with the @WebFilter annotation that has a mapping contained in it.

So, as the origin is used programmatically, I don't want to fiddle around with pretty-printing it as it makes it more difficult to a) calculate and b) use.

Reply:
This is more than pretty printing.   hashCodes are alwayrs reported in hex everywhere else in the code.   If we are ever debugging this, then having the hex rather than the long would be helpful.  Don't care so much about the `@` vs the `.` but the former is more common with hashcodes.

I don't see how this makes it harder to a) calculate and b) use ?

Reply:
Done.